### PR TITLE
hv: virq: refine acrn_handle_pending_request() 

### PR DESCRIPTION
**Pull request by taoyuhong**
hv: virq: refine acrn_handle_pending_request() has more than one exit point
The MISRA-C Standards suggests procedures to be single exit.

hv: virq: refine acrn_handle_pending_request() use goto instruction
The MISRA-C Standards suggests not to use 'goto' instruction.

Tracked-On: #861
Acked-by: Eddie Dong <eddie.dong@intel.com>
Signed-off-by: Tao Yuhong <yuhong.tao@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2171<br>5 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/virq.c&nbsp;&nbsp;function:None&nbsp;&nbsp;offset:382:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Basic type declaration used. : int acrn_inject_pending_vector<br><br>2.filename:/hypervisor/arch/x86/virq.c&nbsp;&nbsp;function:acrn_inject_pending_vector&nbsp;&nbsp;offset:483:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>3.filename:/hypervisor/arch/x86/virq.c&nbsp;&nbsp;function:acrn_inject_pending_vector&nbsp;&nbsp;offset:483:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Basic type declaration used. : int acrn_inject_pending_vector<br><br>4.filename:/hypervisor/arch/x86/virq.c&nbsp;&nbsp;function:acrn_inject_pending_vector&nbsp;&nbsp;offset:485:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Basic type declaration used. : int ret<br><br>5.filename:/hypervisor/arch/x86/virq.c&nbsp;&nbsp;function:acrn_inject_pending_vector&nbsp;&nbsp;offset:512:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Logical conjunctions need brackets.<br><br>

File: hypervisor/arch/x86/virq.c [hunk 1]
```diff
@@ -394,100 +394,89 @@ int32_t acrn_handle_pending_request(struct acrn_vcpu *vcpu)
 	if (bitmap_test_and_clear_lock(ACRN_REQUEST_TRP_FAULT,
```
Comment (lifeix) [inline hypervisor/arch/x86/virq.c]: Could u change these into one line ?

File: hypervisor/arch/x86/virq.c [hunk 2]
```diff
@@ -394,100 +394,89 @@ int32_t acrn_handle_pending_request(struct acrn_vcpu *vcpu)
...
 						pending_req_bits)) {
 		pr_fatal("Triple fault happen -> shutdown!");
-		return -EFAULT;
-	}
-
-	if (bitmap_test_and_clear_lock(ACRN_REQUEST_EPT_FLUSH,
-						pending_req_bits)) {
-		invept(vcpu);
-	}
+		ret =  -EFAULT;
```
Comment (lifeix) [inline hypervisor/arch/x86/virq.c]: more space here.

Comment (taoyuhong) [reply hypervisor/arch/x86/virq.c]: Yes, I will fix this

File: hypervisor/arch/x86/virq.c [hunk 3]
```diff
@@ -534,6 +491,53 @@ int32_t acrn_handle_pending_request(struct acrn_vcpu *vcpu)
 	return ret;
 }
 
+static inline int acrn_inject_pending_vector(struct acrn_vcpu *vcpu, uint64_t *pending_req_bits)
```
Comment (lifeix) [inline hypervisor/arch/x86/virq.c]: Use int32_t instead of int

Comment (taoyuhong) [reply hypervisor/arch/x86/virq.c]: OK

**Comment by HuiHuangShi** [comment]
2171<br>1 new potential violations to the coding guideline detected. Please help check. Thanks!  <br><br>1.filename:/hypervisor/arch/x86/virq.c&nbsp;&nbsp;function:acrn_inject_pending_vector&nbsp;&nbsp;offset:483:<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reason:Pointer param should be declared pointer to const. : vcpu<br><br>

**Pull request merged**